src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts

```diff
@@ -162,13 +161,14 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 		if (!command.marker) {
 			throw new Error(`cannot add a decoration for a command ${JSON.stringify(command)} with no marker`);
 		}
+		this._placeholderDecoration?.dispose();
```

Comment:
anytime a new decoration is registered, we want to dispose of the placeholder.

Reply:
Let's also set to undefined to free the now invalid decoration